frontend/lib/src/dataframes/arrowFormatUtils.ts

```diff
@@ -339,85 +485,47 @@ export function format(x: DataType, type?: Type, field?: Field): string {
   // date
   const isDate = x instanceof Date || Number.isFinite(x)
   if (isDate && typeName === "date") {
-    return moment.utc(x as Date | number).format("YYYY-MM-DD")
+    return formatDate(x as Date | number)
   }
+
   // time
   if (typeof x === "bigint" && typeName === "time") {
     return formatTime(Number(x), field)
   }
 
-  // datetimetz
-  if (isDate && typeName === "datetimetz") {
-    const meta = type?.meta
-    let datetime = moment(x as Date | number)
-
-    if (meta?.timezone) {
-      if (moment.tz.zone(meta?.timezone)) {
-        // uses timezone notation
-        datetime = datetime.tz(meta?.timezone)
-      } else {
-        // uses UTC offset notation
-        datetime = datetime.utcOffset(meta?.timezone)
-      }
-    }
-
-    return datetime.format("YYYY-MM-DD HH:mm:ssZ")
-  }
-  // datetime, datetime64, datetime64[ns], etc.
-  if (isDate && typeName?.startsWith("datetime")) {
-    return moment.utc(x as Date | number).format("YYYY-MM-DD HH:mm:ss")
-  }
-
-  if (typeName?.startsWith("interval")) {
-    return formatIntervalType(x as StructRow, typeName as IntervalType)
+  // datetimetz, datetime, datetime64, datetime64[ns], etc.
+  if (
+    isDate &&
+    (typeName?.startsWith("datetime") || fieldType instanceof Timestamp)
+  ) {
+    return formatDatetime(x as Date | number, field)
   }
 
-  if (typeName?.startsWith("period")) {
-    return formatPeriodType(x as bigint, typeName as PeriodType)
+  if (typeName?.startsWith("period") || extensionName === "pandas.period") {
+    return formatPeriod(x as bigint, field)
   }
 
-  if (typeName === "categorical") {
-    return formatCategoricalType(
-      x as number | bigint | StructRow,
-      field as Field
-    )
+  if (typeName?.startsWith("interval")) {
```

Comment:
for `period` we check also the `extensionName`. Why don't we have to check for `extensionName === "pandas.interval"` here?

Reply:
Yep, I think we can add `extensionName === "pandas.interval"` here as well for now. Both are arrow extension types implemented in pandas. At the moment, the typename check would also cover most/all cases, but that's something that will be refactored in a subsequent PR -> and than we have to mostly rely on the extension type name.